tool/CMakeLists.txt

```diff
@@ -29,17 +31,62 @@ endif()
 
 if(APPLE OR WIN32 OR ANDROID)
   target_link_libraries(bssl ssl crypto)
+  set(LIBRT_FLAGS "")
 else()
   find_library(FOUND_LIBRT rt)
   if(FOUND_LIBRT)
     target_link_libraries(bssl ssl crypto -lrt)
+    set(LIBRT_FLAGS "-lrt")
   else()
     target_link_libraries(bssl ssl crypto)
+    set(LIBRT_FLAGS "")
   endif()
 endif()
 
-install(TARGETS bssl
-        EXPORT bssl-targets
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-)
+function(build_benchmark target_name additional_include_dir lib_crypto)
+  message(${additional_include_dir})
+  add_executable(
+          ${target_name}
+          speed.cc
+          args.cc
+          const.cc
+          benchmark.cc
+  )
+  # Link with the internal tool directory for shared headers with the rest of the tool instead of the top level AWS-LC
+  # include directory
+  target_include_directories(${target_name} PUBLIC ${additional_include_dir} ${AWSLC_INSTALL_DIR}/include/internal/tool)
+  target_link_libraries(${target_name} ${lib_crypto} ${LIBRT_FLAGS})
+
+  string(TOUPPER ${target_name} upper_target_name)
+  target_compile_options(${target_name} PUBLIC -D${upper_target_name})
+endfunction()
+
+if(AWSLC_INSTALL_DIR)
+  build_benchmark(awslc_bm ${AWSLC_INSTALL_DIR}/include crypto)
+endif()
+
+# This expects a directory which contains lib/libcrypto.a and the includes in include/openssl/*.h
+if(OPENSSL_INSTALL_DIR)
+  build_benchmark(ossl_bm ${OPENSSL_INSTALL_DIR}/include ${OPENSSL_INSTALL_DIR}/build/lib/libcrypto.a)
+
+  # libpthread is a linux only library
+  if(NOT MSVC AND NOT ANDROID)
+    # OpenSSL uses pthread_rwlock_unlock from pthread and  dlopen/close from dl
```

Comment:
Quite a bit more symbols than that.

Reply:
Do you know any references I can look at to find the others?

Reply:
I think we just need to link against pthread and the linker lib. No need to specify the actual functions used.

Also, if you put a list with function names here, it would definitely become out-of-date at some point.